src/Microsoft.Identity.Client/Platforms/Android/AuthenticationContinuationHelper.cs

```diff
@@ -52,42 +52,31 @@ private static AuthorizationResult ProcessFromEmbeddedWebview(int requestCode, R
         {
             switch ((int)resultCode)
             {
-                case (int)Result.Ok:
-                    return new AuthorizationResult(AuthorizationStatus.Success, data.GetStringExtra("ReturnedUrl"));
+            case (int)Result.Ok:
+                return AuthorizationResult.FromUri(data.GetStringExtra("ReturnedUrl"));
 
-                case (int)Result.Canceled:
-                    return new AuthorizationResult(AuthorizationStatus.UserCancel, null);
 
-                default:
-                    return new AuthorizationResult(AuthorizationStatus.UnknownError, null);
+            case (int)Result.Canceled:
+                return new AuthorizationResult(AuthorizationStatus.UserCancel);
+
+            default:
+                return new AuthorizationResult(AuthorizationStatus.UnknownError);
             }
         }
 
         private static AuthorizationResult ProcessFromSystemWebview(int requestCode, Result resultCode, Intent data)
         {
             switch ((int)resultCode)
             {
-                case AndroidConstants.AuthCodeReceived:
-                    return CreateResultForOkResponse(data.GetStringExtra("com.microsoft.identity.client.finalUrl"));
-
-                case AndroidConstants.Cancel:
-                    return new AuthorizationResult(AuthorizationStatus.UserCancel, null);
-
-                default:
-                    return new AuthorizationResult(AuthorizationStatus.UnknownError, null);
-            }
-        }
+            case AndroidConstants.AuthCodeReceived:
```

Comment:
nit: Not sure why these are android consts. couldn't these be the same consts across all platforms?